### ISSUE TITLE
cc1: note: obsolete option -I- used, please use -iquote instead

### ISSUE THREAD

**Issue by njdan5691** [att/ast#20]
This is an nmake issue, when nmake probe's the gcc compile it still used the -I- notation. Is there a work around for this?

**Comment by dannyweldon** [comment]
It looks like -iquote was added in GCC 4.0.0, so one would have to assume that no-one is using GCC 3.x.x before changing this.  Does anybody know if GCC 3.x.x is being used?  Also, -iquote doesn't look like a direct swap as -I- is a separator, whereas -iquote takes only a single argument, so the nmake rules in lib/make/Makerules.mk would need some reworking.

**Comment by njdan5691** [comment]
Yes that is the issue -iquote is not really a replacement for -I-. Not sure if there is a better place to report an nmake issue.

**Comment by jhfrontz** [comment]
This is actually a problem with GCC -- when They (GCC developers) first eliminated the -I- option (sometime back in the mid 90s?), They had no idea what the option actually did (and erroneously thought that the "-iquote" fully implemented "-I-").  My recollection is that They forced the change through and it broke lots (and lots and lots) of nmake builds in AT&T/Bell Labs.  So They reluctantly agreed to restore the functionality -- but not without extracting Their pound of flesh (They added the "note: obsolete option -I- used, please use -iquote instead" message that we nmake users have had to live with for 20+ years).

So, it's not an nmake issue; it's a GCC issue.

**Comment by njdan5691** [comment]
Thanks jhfrontz.

**Comment by dannyweldon** [comment]
FYI.  I just ran a small test using gcc 4.9.2 and even using -Wno-deprecated didn't suppress it.

**Comment by jhfrontz** [comment]
This says that it happened in mid-2000s, but I swear that we had the same problem back in the mid-90s.  So maybe the 21st-century attempt was a repeat?

http://nmake.alcatel-lucent.com/newsletters/issue021.html#gcc

**Comment by krader1961** [comment]
Note that the use of `-I-` makes it impossible to use `clang` (the LLVM frontend) since it treats the flag as an error. I hacked the `ppcc` script to strip that flag from the args which allows using clang to build this project. There's no point changing Nmake at this juncture since it would be more productive to switch to Cmake or Meson as the build system. See issue #42.

**Comment by jhfrontz** [comment]
Is there anything in clang that implements the functionality of `-I-` (which I think is used to tell the compiler to forget about using the stock #include search path)?

**Comment by krader1961** [comment]
This [clang CLI document](https://clang.llvm.org/docs/ClangCommandLineReference.html#include-path-management) implies the flag is still legal. So apparently the implementation was changed recently and the online documentation hasn't caught up. Replacing `-I-` with the long form `--incude-barrier` results in the same error (not surprisingly). But note that at least on Linux and macOS the flag isn't even needed when compiling with clang. I hacked the `ppcc.sh` script to simply remove that flag from the list of arguments and can successfully build everything.
